### PR TITLE
Active admin

### PR DESCRIPTION
**Pull request by agustinbajo**
[https://trello.com/c/3RyIH0lv](https://trello.com/c/3RyIH0lv)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,6 +7,13 @@
 # Ignore bundler config.
 /.bundle
 
+
+#Ignore secrets
+/secrets.yml
```
Comment (holywyvern) [inline .gitignore]: why is secrets.yml ignored?
what does it contains?

File: Gemfile [hunk 2]
```diff
@@ -48,6 +53,9 @@ group :development, :test do
   gem 'shoulda-matchers', '~> 3.1.2'
   # Use sqlite3 as the database for Active Record
   gem 'sqlite3'
+  gem 'dotenv-rails', groups: [:development, :test]
+  gem 'activeadmin' , '~> 1.3.1'
+  gem 'pundit', '~> 2.0.0'
```
Comment (holywyvern) [inline Gemfile]: Pundit should be on all workspaces, not only development and testing.

File: app/admin/dashboard.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  content title: proc{ I18n.t("active_admin.dashboard") } do
+    div class: "blank_slate_container", id: "dashboard_default_message" do
+      span class: "blank_slate" do
+        span I18n.t("active_admin.dashboard_welcome.welcome")
+        small I18n.t("active_admin.dashboard_welcome.call_to_action")
+      end
+    end
+
+    # Here is an example of a simple dashboard with columns and panels.
```
Comment (holywyvern) [inline app/admin/dashboard.rb]: remove code comments.

File: app/controllers/api/v1/books_controller.rb [hunk 4]
```diff
@@ -10,7 +10,7 @@ def index
 
       def show
         @book = Book.find(params[:id])
-        render json: @book
+        render_paginated json: @book
```
Comment (holywyvern) [inline app/controllers/api/v1/books_controller.rb]: `Book.find(id)` always return a single value, why did you use pagination on it ?

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -3,7 +3,7 @@ class ApplicationController < ActionController::Base
 
   protect_from_forgery with: :null_session
   before_action :configure_permitted_parameters, if: :devise_controller?
-
+  skip_before_action :verify_authenticity_token #todo
```
Comment (holywyvern) [inline app/controllers/application_controller.rb]: This is OK, but add a comment explaining why you had to do this above. Remember, todo is always in caps and with a description:

```rb
# TODO, review why token is not working for an API.
skip_before_action :verify_authenticity_token
```

File: app/mailers/application_mailer.rb [hunk 6]
```diff
@@ -1,4 +1,4 @@
 class ApplicationMailer < ActionMailer::Base
-  default from: 'from@example.com'
+  default from: 'from_email'
```
Comment (holywyvern) [inline app/mailers/application_mailer.rb]: this should be something like `Rails.application.secrets.from_email` instead. Or remove the entire line.

File: app/mailers/user_mailer.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+class UserMailer < ApplicationMailer
+  default from: Rails.application.secrets.from_email
```
Comment (holywyvern) [inline app/mailers/user_mailer.rb]: If you kept the line in ApplicationMailer you should remove it from here.
If you remove it from ApplicationMailer, you then keep this one instead.

File: app/views/user_mailer/welcome_email.html.erb [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
+    <h1><%= "Ruby on Rails Training"%></h1>
+    <p>
+    <b><%= t I18n.t('.email.created_at') %> </b> <%= @rent.created_at %><br>
+      <b><%= t I18n.t('.book.details') %> </b>
+      <br>
+          <b><%= t  I18n.t('.book.gender') %> </b><%= @book.gender %><br>
```
Comment (holywyvern) [inline app/views/user_mailer/welcome_email.html.erb]: remember to remove the `I18n.t` and just keep it as t 'key'.

Also, keys should be without an starting dot:

`<%= t I18n.t('.email.created_at') %> ` should be `<%= t 'email.created_at' %> ` instead.

File: config/environments/production.rb [hunk 9]
```diff
@@ -79,6 +79,8 @@
   # Use a different logger for distributed setups.
   # require 'syslog/logger'
   # config.logger = ActiveSupport::TaggedLogging.new(Syslog::Logger.new 'app-name')
+  config.action_mailer.delivery_method = :smtp  # liena nueva
+  config.action_mailer.perform_deliveries = false # liena nueva
```
Comment (holywyvern) [inline config/environments/production.rb]: remove this line.

File: spec/mailers/user_spec.rb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+require "rails_helper"
```
Comment (holywyvern) [inline spec/mailers/user_spec.rb]: you may remove this file.

File: spec/models/admin_user_spec.rb [hunk 11]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
```
Comment (holywyvern) [inline spec/models/admin_user_spec.rb]: this file should be removed too.

**Review by holywyvern — CHANGES_REQUESTED** [review]

**Comment by holywyvern** [comment]
As a side note, when you make a git commit, remember to follow the standar:

```markdown
### Summary

- Changes you made
- Commented as a list like this

### Screenshots

You won't usually do this, unless you make something in frontend. But it is usually there.

### Trello card

Link to trello card

### Notes

You may add additional notes here, in case you need

```

This is an example of a good pull request:

[PR good practices](https://github.com/Wolox/tech-guides/blob/master/git/docs/pull-requests-good-practices.png?raw=true)

File: spec/models/admin_user_spec.rb (hunk 11 above)
Comment (agustinbajo) [reply spec/models/admin_user_spec.rb]: done!

File: spec/mailers/user_spec.rb (hunk 10 above)
Comment (agustinbajo) [reply spec/mailers/user_spec.rb]: done!

File: config/environments/production.rb (hunk 9 above)
Comment (agustinbajo) [reply config/environments/production.rb]: done!

File: app/views/user_mailer/welcome_email.html.erb (hunk 8 above)
Comment (agustinbajo) [reply app/views/user_mailer/welcome_email.html.erb]: done!

File: app/controllers/application_controller.rb (hunk 5 above)
Comment (agustinbajo) [reply app/controllers/application_controller.rb]: done!

File: app/controllers/api/v1/books_controller.rb (hunk 4 above)
Comment (agustinbajo) [reply app/controllers/api/v1/books_controller.rb]: done!

File: app/admin/dashboard.rb (hunk 3 above)
Comment (agustinbajo) [reply app/admin/dashboard.rb]: done!

File: Gemfile (hunk 2 above)
Comment (agustinbajo) [reply Gemfile]: done!

File: app/controllers/api/v1/api_controller.rb [hunk 12]
```diff
@@ -0,0 +1,22 @@
+dule Api
```
Comment (holywyvern) [inline app/controllers/api/v1/api_controller.rb]: shouldn't this be `module Api` ?

**Review by holywyvern — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/api_controller.rb (hunk 12 above)
Comment (agustinbajo) [reply app/controllers/api/v1/api_controller.rb]: Fixed

**Review by holywyvern — APPROVED** [review]
Well done!

**Pull request merged**